### ISSUE TITLE
error in mcmc_IMIFA when range.Q includes 0

### ISSUE THREAD

**Issue by abcoxyzide** [Keefe-Murphy/IMIFA-GIT#1]
Thanks for the cool work!

In the documentation, it is written that 
> "...for methods not ending in IFA it is possible to fit zero-factor models, corresponding to simple diagonal covariance structures."

My test code:

    set.seed(123)
    sim <- data.frame(
        x = c(
            rnorm(900, mean = 0, sd = 0.2),
            rnorm(100, mean = 0.5, sd = 0.2)
        )
    )
    fit1 <- mcmc_IMIFA(sim, method = 'MFA', range.G = 2, range.Q = 0)
    fit2 <- mcmc_IMIFA(sim, method = 'IMFA', range.Q = 0)

Both `fit1` and `fit2` returns an error saying `in eta[z == g, , drop = FALSE] : (subscript) logical subscript too long`

I am using the github version of `IMIFA`.

**Comment by Keefe-Murphy** [comment]
Thanks for your interest in the package! I'm sorry for the extremely delayed reply; unfortunately, the notification email went straight to my spam folder.

Anyway, it looks like (unbenknowst to me) I broke this feature quite some time ago in an attempt to make other things a little faster. I haven't committed the changes or submitted to CRAN yet, as I'm still working on other edits, but the attached file should work as originally advertised for FA/MFA/OMFA/IMFA models. At least, I verified that it does on your test code. Somehow there was no such problem when the infinite factor models adaptively truncated to zero factors...

[IMIFA_2.1.11.tar.gz](https://github.com/Keefe-Murphy/IMIFA/files/11806434/IMIFA_2.1.11.tar.gz)

Cheers, and thanks again for flagging this!
